src/ape/managers/chain.py

```diff
@@ -0,0 +1,144 @@
+from typing import List, Optional
+
+from dataclassy import dataclass
+
+from ape.api import BlockAPI, ProviderAPI, TestProviderAPI
+from ape.exceptions import ChainError, ProviderNotConnectedError, UnknownSnapshotError
+from ape.types import SnapshotID
+
+from .networks import NetworkManager
+
+
+@dataclass
+class BlockContainer:
+    """
+    A list of blocks on the chain.
+
+    Usages example::
+
+        from ape import chain
+
+        latest_block = chain.blocks[-1]
+
+    """
+
+    _provider: ProviderAPI
+
+    def __getitem__(self, block_number: int) -> BlockAPI:
+        """
+        Get a block by number. Negative numbers start at the chain head and
+        move backwards. For example, ``-1`` would be the latest block and
+        ``-2`` would be the penultimate block, and so on.
```

Comment:
Penultimate is a little too fancy for some readers lol
```suggestion
        ``-2`` would be the block prior to that one, and so on.
```

Reply:
ya penultimate almost sounds like it is the second to last block EVER too